File: php/src/MBPay/Types/PaymentLinkRequest.php
<?php

namespace MBPay\Types;

/**
 * 支付链接生成请求参数
 */
class PaymentLinkRequest
{
    /**
     * @var string 商户订单号（必填）
     */
    private $orderNo;

    /**
     * @var string 商品描述（必填）
     */
    private $subject;

    /**
     * @var int 订单金额（分，必填）
     */
    private $amount;

    /**
     * @var int 过期时间（分钟，必填）
     */
    private $expire;

    /**
     * @var string|null 随机数（可选，不填会自动生成）
     */
    private $nonce;

    /**
     * @var string|null 回调通知地址（可选）
     */
    private $notifyUrl;

    /**
     * @param string $orderNo 商户订单号（必填）
     * @param string $subject 商品描述（必填）
     * @param int $amount 订单金额（分，必填）
     * @param int $expire 过期时间（分钟，必填）
     * @param string|null $nonce 随机数（可选，不填会自动生成）
     * @param string|null $notifyUrl 回调通知地址（可选）
     */
    public function __construct(
        string $orderNo,
        string $subject,
        int $amount,
        int $expire,
        ?string $nonce = null,
        ?string $notifyUrl = null
    ) {
        $this->orderNo = $orderNo;
        $this->subject = $subject;
        $this->amount = $amount;
        $this->expire = $expire;
        $this->nonce = $nonce;
        $this->notifyUrl = $notifyUrl;
    }

    /**
     * 获取商户订单号
     *
     * @return string
     */
    public function getOrderNo(): string
    {
        return $this->orderNo;
    }

    /**
     * 获取商品描述
     *
     * @return string
     */
    public function getSubject(): string
    {
        return $this->subject;
    }

    /**
     * 获取订单金额
     *
     * @return int
     */
    public function getAmount(): int
    {
        return $this->amount;
    }

    /**
     * 获取过期时间
     *
     * @return int
     */
    public function getExpire(): int
    {
        return $this->expire;
    }

    /**
     * 获取随机数
     *
     * @return string|null
     */
    public function getNonce(): ?string
    {
        return $this->nonce;
    }

    /**
     * 获取回调通知地址
     *
     * @return string|null
     */
    public function getNotifyUrl(): ?string
    {
        return $this->notifyUrl;
    }
}

File: java/src/main/java/com/mbpay/model/PaymentLinkRequest.java
package com.mbpay.model;

/**
 * 支付链接生成请求参数
 */
public class PaymentLinkRequest {
    private String orderNo;   // 商户订单号（必填）
    private String subject;   // 商品描述（必填）
    private long amount;      // 订单金额（分，必填）
    private int expire;       // 过期时间（分钟，必填）
    private String nonce;     // 随机数（可选，不填会自动生成）
    private String notifyUrl; // 回调通知地址（可选）

    /**
     * @param orderNo 商户订单号（必填）
     * @param subject 商品描述（必填）
     * @param amount 订单金额（分，必填）
     * @param expire 过期时间（分钟，必填）
     * @param nonce 随机数（可选，不填会自动生成）
     * @param notifyUrl 回调通知地址（可选）
     */
    public PaymentLinkRequest(String orderNo, String subject, long amount, int expire, String nonce, String notifyUrl) {
        this.orderNo = orderNo;
        this.subject = subject;
        this.amount = amount;
        this.expire = expire;
        this.nonce = nonce;
        this.notifyUrl = notifyUrl;
    }

    public PaymentLinkRequest(String orderNo, String subject, long amount, int expire) {
        this(orderNo, subject, amount, expire, null, null);
    }

    public String getOrderNo() {
        return orderNo;
    }

    public String getSubject() {
        return subject;
    }

    public long getAmount() {
        return amount;
    }

    public int getExpire() {
        return expire;
    }

    public String getNonce() {
        return nonce;
    }

    public String getNotifyUrl() {
        return notifyUrl;
    }
}






